src/main/jhipster/src/main/webapp/app/germplasm-manager/germplasm-import-update-dialog.component.ts

```diff
@@ -53,18 +63,23 @@ export class GermplasmImportUpdateDialogComponent implements OnInit, OnDestroy {
         this.activeModal.dismiss('cancel');
     }
 
-    import() {
+    async import() {
         this.isProcessing = true;
-        this.validate().subscribe((isValid) => {
-            if (isValid) {
-                this.germplasmService.importGermplasmUpdates(this.transform(this.data, this.names, this.attributes)).subscribe(
-                    (res: HttpResponse<any>) => this.onSaveSuccess(res.body),
-                    (res: HttpErrorResponse) => this.onError(res)
-                );
-            } else {
-                this.isProcessing = false;
-            }
-        });
+        const isValid = await this.validate();
+        if (!isValid) {
+            this.isProcessing = false;
+            return;
+        }
+        try {
+            await this.confirmAttributeStatus();
+        } catch (e) {
+            this.isProcessing = false;
+            return;
+        }
+        this.germplasmService.importGermplasmUpdates(this.transform(this.data, this.names, this.attributes)).subscribe(
+            (res: HttpResponse<any>) => this.onSaveSuccess(res.body),
+            (res: HttpErrorResponse) => this.onError(res)
+        );
```

Comment:
@abatac I refactored the  germplasm-import-update-dialog.component `import()` and
`validate()` methods using async/await syntax, to simplify adding more asynchronous confirmation steps. Let me know if you have any concerns